README.md

```diff
@@ -1,107 +1,149 @@
 # Overview
 
-[![Join the chat at https://gitter.im/BinaryAnalysisPlatform/bap](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/BinaryAnalysisPlatform/bap?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) [![Build Status](https://travis-ci.org/BinaryAnalysisPlatform/bap.svg?branch=master)](https://travis-ci.org/BinaryAnalysisPlatform/bap) [![docs](https://img.shields.io/badge/doc-v0.9.9-green.svg)](http://binaryanalysisplatform.github.io/bap/api/v0.9.9/Bap.Std.html) [![docs](https://img.shields.io/badge/doc-master-green.svg)](http://binaryanalysisplatform.github.io/bap/api/master/Bap.Std.html)
-
-BAP is a platform for binary analysis. It is written in OCaml, but can
-be used from other languages.
+[![Join the chat at https://gitter.im/BinaryAnalysisPlatform/bap](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/BinaryAnalysisPlatform/bap?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+[![docs](https://img.shields.io/badge/doc-1.0.0-green.svg)](http://binaryanalysisplatform.github.io/bap/api/v1.0.0/argot_index.html)
+[![docs](https://img.shields.io/badge/doc-master-green.svg)](http://binaryanalysisplatform.github.io/bap/api/master/argot_index.html)
+[![Build Status](https://travis-ci.org/BinaryAnalysisPlatform/bap.svg?branch=master)](https://travis-ci.org/BinaryAnalysisPlatform/bap)
+[![pip](https://img.shields.io/badge/pip-1.1.0-green.svg)](https://pypi.python.org/pypi/bap/)
+[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/BinaryAnalysisPlatform/bap/blob/master/LICENSE)
+
+Binary Analysis Platform is a framework for writing program analysis
+tools, that target binary files. The framework consists of a bunch
+of libraries, plugins and frontends. The libraries provide code
+reusability, plugins facilitate extensibility and frontends serve as
+entry points.
 
 # <a name="Installation"></a>Installation
 
-BAP is released using `opam` package manager. After you've successfully
+We use OPAM package manager to handle installation. After you've successfully
 [installed](https://opam.ocaml.org/doc/Install.html) opam, do the following:
 
 ```bash
-$ opam init --comp=4.02.1
-$ eval `opam config env`
-$ opam install depext
-$ opam depext bap
-$ opam install bap
+$ opam init --comp=4.02.3    # install the compiler
+$ opam repo add bap git://github.com/BinaryAnalysisPlatform/opam-repository
+$ eval `opam config env`               # activate opam environment
+$ opam depext --install bap            # install bap
 ```
 
+
 # Usage
```

Comment:
Maybe the Shell section should be moved up here? Using it from OCaml involves using the shell frontend to some extent - it might make more sense to a user if they trigger a provided plugin before writing their own.